Core/Utility/ManifestUtility.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using NuGet.Versioning;
+
+namespace NuGetPe.Utility
+{
+    public static class ManifestUtility
+    {
+        const string TokenMetadataStart = "0.0.0-TOKENSTART.";
+        const string TokenMetadataEnd = ".TOKENEND";
+        static readonly Regex tokenRegex = new Regex(@"([$])(?:(?=(\\?))\2.)*?\1", RegexOptions.Compiled);
+        static readonly Regex metadataRegEx = new Regex(@"0\.0\.0\-TOKENSTART\.([^.]+)\.TOKENEND", RegexOptions.Compiled);
```

Comment:
do we really need a regex? I prefer `IndexOf` for simplicity and more clear performance.

Reply:
The RegEx was the easiest to get the entire token span, start to end, also accounting for multiple token values in a string (which is valid for the VersionRange on a dependency group).

I didn't see any perf issues here as it's also a UI app, not a server handling tons of requests :)